fix(ci): iOS deploy action.yml YAML heredoc 파싱 에러 — 4일 연속 배포 실패

Scheduler: tpm-exec-report

### 배경
2026-03-27부터 iOS User/Partner 배포가 모두 실패 중 (4일, 8회 연속). PR #749에서 reusable workflow → composite action 변환 시 도입된 YAML 파싱 버그가 root cause.

**출처**: TPM 운영 분석 (기존 report-exec #702의 근본 원인 규명)

### 수정 대상
| 파일 | 라인 | 문제 | 수정 방향 |
|------|------|------|----------|
| `.github/actions/ios-deploy/action.yml` | L198-203 | shell heredoc 콘텐츠가 YAML `run: \|` 블록 인덴트 밖에 위치 → YAML 파서가 `CODE_SIGN_IDENTITY = Apple Distribution`의 `:`를 YAML 키로 해석하여 파싱 실패 | heredoc 대신 `echo` + `>>` 또는 `printf` 사용 |

### 에러 메시지
```
##[error].github/actions/ios-deploy/action.yml: (Line: 200, Col: 1, Idx: 7612) - While scanning a simple key, could not find expected ':'.
##[error]Failed to load .github/actions/ios-deploy/action.yml
```

### 구현 가이드
1. `.github/actions/ios-deploy/action.yml` L198-203의 heredoc을 제거
2. `echo` 또는 `printf` 방식으로 변경:
```yaml
      run: |
        # ... (기존 코드)
        
        # Create xcconfig with correct settings
        {
          echo "CODE_SIGN_STYLE = Manual"
          echo "CODE_SIGN_IDENTITY = Apple Distribution"
          echo "DEVELOPMENT_TEAM = ${{ inputs.development-team }}"
          echo "PROVISIONING_PROFILE_SPECIFIER = $PROFILE_SPECIFIER"
        } > ios/CI.xcconfig
        
        # ... (이후 코드 동일)
```
3. 같은 파일에 다른 heredoc이 있는지 확인하고 동일 패턴 수정

### 검증 방법
- iOS User App Deploy / iOS Partner App Deploy 워크플로우 성공 확인
- `act` 로컬 테스트 또는 push 후 CI 확인

--- a/.github/actions/ios-deploy/action.yml
+++ b/.github/actions/ios-deploy/action.yml
@@ -195,12 +195,13 @@ runs:
         fi
 
         # Create xcconfig with correct settings
-        cat > ios/CI.xcconfig << EOF
-CODE_SIGN_STYLE = Manual
-CODE_SIGN_IDENTITY = Apple Distribution
-DEVELOPMENT_TEAM = ${{ inputs.development-team }}
-PROVISIONING_PROFILE_SPECIFIER = $PROFILE_SPECIFIER
-EOF
+        # Fix #901: heredoc content was parsed as YAML keys due to ':'
+        {
+          echo "CODE_SIGN_STYLE = Manual"
+          echo "CODE_SIGN_IDENTITY = Apple Distribution"
+          echo "DEVELOPMENT_TEAM = ${{ inputs.development-team }}"
+          echo "PROVISIONING_PROFILE_SPECIFIER = $PROFILE_SPECIFIER"
+        } > ios/CI.xcconfig
 
         # Include in Release.xcconfig
         echo '#include "../CI.xcconfig"' >> ios/Flutter/Release.xcconfig

